k2/csrc/cuda/log_test.cu

```diff
@@ -0,0 +1,63 @@
+// k2/csrc/cuda/log.h
+
+// Copyright (c)  2020  Fangjun Kuang (csukuangfj@gmail.com)
+
+// See ../../../LICENSE for clarification regarding multiple authors
+
+#include "gtest/gtest.h"
+#include "k2/csrc/cuda/log.h"
+
+namespace k2 {
+
+TEST(Log, Cpu) {
+  K2_LOG(DEBUG) << "Debug message";
+  K2_LOG(INFO) << "Info message";
+  K2_LOG(WARNING) << "Warning message";
+  K2_LOG(ERROR) << "Error message";
+
+  int a = 10;
+  int b = 20;
+  int c = 30;
+  int d = a;
+
+  K2_CHECK(a == d) << "failed";
+
+  K2_CHECK_EQ(a, d) << "failed";
+  K2_CHECK_NE(a, b) << "failed";
+
+  K2_CHECK_LE(a, b) << "failed";
+  K2_CHECK_LT(a, b) << "failed";
+
+  K2_CHECK_GE(c, b) << "failed";
+  K2_CHECK_GT(c, b) << "failed";
+}
+
+__global__ void DummyKernel(int *b, int a) {
+  K2_LOG(INFO) << "In kernel";
+  K2_CHECK_LT(*b, a) << K2_KERNEL_DEBUG_STR;
+  *b += 1;
+  K2_CHECK_EQ(*b, a) << K2_KERNEL_DEBUG_STR;
+}
+
+TEST(Log, Cuda) {
+  K2_LOG(INFO) << "Test log for cuda";
+  int a = 10;
+  int *b;
+  auto ret = cudaMalloc(&b, sizeof(a));
+  K2_CHECK_EQ(ret, cudaSuccess) << "Failed to allocate memory";
+
+  ret = cudaMemcpy(b, &a, sizeof(a), cudaMemcpyHostToDevice);
+  K2_CHECK_EQ(ret, cudaSuccess) << "Failed to copy memory to gpu";
```

Comment:
would like to define macro `K2_CHECK_CUDA_ERROR` for such usages.

Reply:
... currently, K2_CHECK_CUDA_ERROR is for checking cudaError_t return codes, which I think makes sense..
I'm OK in principle with having different macro names for usage inside kernels (or lambdas) but not sure if it's necessary.

Reply:
It's very simple to support `K2_CHECK_CUDA_ERROR`:

```cpp
#define K2_CHECK_CUDA_ERROR(x) K2_CHECK_EQ(x, cudaSuccess)
```

Not sure if we need another macro.

BTW, I find that the macro name `K2_CHECK_CUDA_ERROR` is too long.

Reply:
> K2_CHECK_CUDA_ERROR does more than that, it prints out the CUDA error
code's name.  (Or if it doesn't, it should).

Done.